src/app.py

```diff
@@ -26,74 +36,56 @@ def get_secret_value(secret_arn: str):
     return response.get('SecretString')
 
 
-def sleep_data(garth_client: Client) -> dict:
+def sleep_data(client: GarthClient) -> dict:
     """
     Retrieves the daily sleep data for the user associated with the provided Garmin client.
 
     Args:
-        garth_client (Client): The Garmin Connect API client.
+        client (GarthClient): The Garmin Connect API client.
 
     Returns:
         dict: The daily sleep data for the user.
     """
-    try:
-        return garth_client.connectapi(
-            f"/wellness-service/wellness/dailySleepData/{garth_client.username}",
-            params={"date": datetime.date.today().isoformat(),
-                    "nonSleepBufferMinutes": 60},
-        )
-    except Exception as e:
-        print(e)
+    return client.connectapi(
```

Comment:
Also this method can return Any | None, make sure you're handling the None from the client or you'll be tearing your hair out when you're trying to `json.dumps(None)` to S3.

https://github.com/matin/garth/blob/main/garth/http.py#L158

Maybe validate the garth response inside the function so you're sure you're always returning a dict with atleast some of the structure you want.